quickwit/quickwit-opentelemetry/src/otlp/mod.rs

```diff
@@ -220,6 +222,21 @@ pub(crate) fn extract_otel_index_id_from_metadata(
     Ok(index_id.to_string())
 }
 
+fn ingest_failures_to_error(mut response: IngestResponseV2) -> Result<(), IngestServiceError> {
```

Comment:
The contract of this function is weird. Why should I expect it to fail if there is one single failure/success?

I'm ok for implementing a helper but it should be clear for the user that it is handling only one subrequest. Adding docs will help too.

Reply:
I factorized a bit more code to get rid of this weird function.